programs/bpf_loader/src/syscalls.rs

```diff
@@ -1169,7 +1165,7 @@ impl<'a> SyscallObject<BpfError> for SyscallMemcpy<'a> {
         memory_mapping: &MemoryMapping,
         result: &mut Result<u64, EbpfError<BpfError>>,
     ) {
-        if check_overlapping(src_addr, dst_addr, n) {
+        if !is_nonoverlapping(src_addr, dst_addr, n) {
```

Comment:
This needs to be feature gated, are you familiar with how to do that?  If so, please use the following for the feature pubkey: 36PRUK2Dz6HWYdG9SpjeAsF5F3KxnFCakA2BZMbtMhSb